designs/containers/vnf-deployment/vnf-deployment.md

```diff
@@ -84,23 +130,202 @@ public class ApplianceSoftwareVersion extends BaseEntity {
 }
 ```  
 
+#### Deployment Spec
+The `DeploymentSpec` domain entity will have some of its fields made `nullable`; the new `externalId` field to store the Kubernetes deployment UID; the `namespace` field for the deployment namespace; and set of the new entity `InspectionPort` :
+```java
+Entity
+@Table(name = "DEPLOYMENT_SPEC", ...)
+public class DeploymentSpec extends BaseEntity implements LastJobContainer {
+
+    // ...
+
+    @Column(name = "region", nullable = **true**)
+    private String region;
+
+    @Column(name = "project_name", nullable = **true**)
+    private String projectName;
+
+    @Column(name = "project_id", nullable = **true**)
+    private String projectId;
+
+    @Column(name = "management_network_name", nullable = **true**)
+    private String managementNetworkName;
+
+    @Column(name = "management_network_id", nullable = **true**)
+    private String managementNetworkId;
+
+    @Column(name = "inspection_network_name", nullable = **true**)
+    private String inspectionNetworkName;
+
+    @Column(name = "inspection_network_id", nullable = **true**)
+    private String inspectionNetworkId;
+	
+	@Column(name = "external_id", nullable = **true**)
+    private String externalId;                           // Stores the Kubernetes deployment id.
+	
+	@Column(name = "namespace", nullable = **true**)
+    private String namespace;                           // Stores the Kuberneted deployment namespace.
+	
+	@OneToMany(mappedBy = "deploymentSpec", fetch = FetchType.LAZY)
+    private Set<InspectionPort> inspectionPorts = new HashSet<InspectionPort>(); // The set of inspection ports associated with this DS.

```

Comment:
We should refer to the inspection port via the DAI. It seems confusing to have ports associated directly with the DS because the DS does not manage ports...it manages instances. Its fine to have a instance with no inspection ports associated since the pod was deleted etc...

Reply:
@arvindn05 As discussed I moved this information to be directly in the DAI and update the sync tasks accordingly.